### ISSUE TITLE
Provide instructions for GitHub's new "fine-grained" personal access tokens

### ISSUE THREAD

**Issue by aheckler** [refined-github/refined-github#6092]
### Description

GitHub recently introduced a new personal access token generator, detailed in this blog post:

https://github.blog/2022-10-18-introducing-fine-grained-personal-access-tokens-for-github/

The page where you can generate a token:

https://github.com/settings/tokens?type=beta

Refined GitHub might want to provide instructions for using this method to generate a token.

### Screenshot

![Screenshot taken on 2022-10-20 at 13 40 39 UTC@2x](https://user-images.githubusercontent.com/5512652/196964631-80609ff3-1532-4726-a120-87fbb3eeff3b.png)

Old token generation page is now labelled "Classic".

### Example URLs

The extension settings page perhaps, or maybe somewhere in this repo?

**Comment by busches** [comment]
It's still in beta and we deep link to the existing  page already. Good to keep an eye on, but probably not something we want to make any changes for now.

**Comment by fregante** [comment]
I was going to open a new issue about it yesterday but I wasn't sure if we need to make any changes at all. For Refined GitHub to work, we need write access to all repos, which seems to still be available. 

I don't particularly want to spend time requesting exclusively the permissions we're currently using and then having to deal with broken new features later. If the user decides to customize the page, they'd do at their own peril/safety

**Comment by mtsknn** [comment]
I was going to generate a new beta token (i.e. a fine-grained token) because my classic token doesn't work anymore in a private organization. Refined GitHub logs this to the console:

> ℹ️ ci-link → GraphQL:
> `redacted_organization_name` forbids access via a personal access token (classic). Please use a GitHub App, OAuth App, or a personal access token with fine-grained permissions.

I don't know what permissions I should choose when generating a beta token, so documentation would be nice.

I anyway tried generating a beta token, but Refined GitHub's options page seems to accept only classic tokens (`ghp_...`), not beta tokens (`github_pat_...`):

https://github.com/refined-github/refined-github/blob/747b87a6358c729e54148021d44c94e828603e01/source/options.html#L22

https://github.com/refined-github/refined-github/blob/747b87a6358c729e54148021d44c94e828603e01/source/options.tsx#L97-L101

**Comment by fregante** [comment]
Thanks for the detailed report, @mtsknn! 

Support for fine-grained tokens should come in the next release.



> my classic token doesn't work anymore in a private organization.

That can be customize in the token itself, I've been using the same classic token for a while and it works even on private repos. See this screenshot taken on https://github.com/refined-github/private/issues/1 with a classic token and with `bugs-tab` working (which requires a token)

<img width="505" alt="Screenshot 7" src="https://github.com/refined-github/refined-github/assets/1402241/adbbb28d-a65e-4767-9650-67d249d57ea0">

**Comment by mtsknn** [comment]
> Support for fine-grained tokens should come in the next release.

Nice!

---

> I've been using the same classic token for a while and it works even on private repos.

My classic token has also worked in private _repos_ so far, but today I noticed that it doesn't work in a repo under a certain private _organization_. So I think it's an organization-level setting; the console logging quoted in my previous comment also says so (or so I interpret it):

> `organization_foo` forbids access via a personal access token (classic). [...]

Organization settings > Personal access tokens (e.g. https://github.com/organizations/refined-github/settings/personal-access-tokens-onboarding) looks like this in one of my own private organizations:

> <img width="952" alt="image" src="https://github.com/refined-github/refined-github/assets/2226144/f21abda2-fb02-4f49-8582-04f96540132e">

Step 3 is "Restrict access via personal access tokens (classic)" – I think my problem stems from this, but I can't check the settings of the problematic organization because I'm not authorized to do that.

(Whoops, a bit off topic maybe.)

**Comment by fregante** [comment]
Update: in most cases, you can continue using the classic tokens without issue. GitHub’s error message with workflow is actually super clear so we don't need to update anything at this point.

This is now purely a documentation issue and I don't see the urgency to document it just yet (also because we'd also have to update the scope-checker widget, etc etc)

**Comment by thealphadollar** [comment]
>This is now purely a documentation issue and I don't see the urgency to document it just yet (also because we'd also have to update the scope-checker widget, etc etc)

Hey @fregante , I think it would be a nice to have update since a lot of organizations have disabled Classic PATs and the extension is unusable in those cases 😢

**Comment by amaisano** [comment]
Is this why I am now unable to use the `(X)` button on labels on my PR? I receive this error:

> Unable to fetch. Ensure that your token has access to this repo. { "message": "`m***` forbids access via a personal access token (classic). **Please use a GitHub App, OAuth App, or a personal access token with fine-grained permissions.**", "documentation_url": "https://docs.github.com/rest/issues/labels#remove-a-label-from-an-issue" }

**Comment by 134130** [comment]
@amaisano It seems. Your organization may forbids Classic token.

**Comment by fregante** [comment]
Fine-grained tokens are ultimately not compatible with Refined GitHub at the moment. 

- Classic tokens give access to the same repositories a user has access to, but fine-grained are scoped per-user/per-organization. 
- Supporting multiple tokens would requires a lot of code changes around the options page and API calls, it's not going to happen until GitHub sunsets classic tokens
- You literally cannot create tokens for some organizations, for example I'm part of 8 organizations but can only create tokens for 2

	<img width="335" alt="Screenshot 11" src="https://github.com/refined-github/refined-github/assets/1402241/336969df-386c-4f8e-8f47-7330747fcd79">

- It looks like there's no URL shortcut to pre-fill the form, like there used to be

	https://github.com/refined-github/refined-github/blob/e9e650da9918a1af25c36c59249badfe3f8e0508/source/options.html#L20


In short:

- If your organization does not support classic tokens, then Refined GitHub can only **read public data** from those organizations
- If you want to use a fine-grained token, feel free to do so, but we're not going to keep a documentation page with all the (ever-changing) scope needs required. If you want to limit access to Refined GitHub, you will have to go through each scope anyway. 
- Note that we don't have access to the token nor do we ever collect any data, so you're effectively only limiting your own machine.

In the future:

- If GitHub completely sunsets classic tokens, we can reconsider then. Maybe switch to a hosted GitHub App
- #6951
- #6952

Please open new issues if you encounter specific bugs with fine-grained tokens

**Comment by mdbhm502** [comment]
أريد تصحيح جزئي بكامله وبثون وجميل موبايل 

<details>
<summary>Details</summary>



</details>
